src/unspread-lists.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { visit, CONTINUE } from 'unist-util-visit';
+
+/**
+ * Collapse (un-spread) all lists
+ * @param {object} tree
+ */
+export function unspreadLists(tree, url, log) {
+  let listsWithSpread = 0;
+  let listItemsWithMultipleParagraphs = 0;
+
+  visit(tree, (node) => {
+    if (node.type === 'list' || node.type === 'listItem') {
+      // eslint-disable-next-line no-param-reassign
+      // node.spread = false;
+      if (node.spread) {
+        listsWithSpread += 1;
+      }
+    }
+    if (node.type === 'listItem' && node.children.length > 1) {
+      // unwrap paragraphs if multiple children
+      // const children = [];
+      // node.children.forEach((child) => children.push(...child.children));
```

Comment:
wouldn't that be wrong for non-paragraph childeren? eg `<li><code>foo</code>`

Reply:
Right. I added roundtrip tests for various combinations of code (inline, blocks, with and without other text content). There is only the 2 cases that wrap the text wrongly:

- picture + code block
- code block + text + code block (spread)

In fact even `<li><picture></li>` (a single child) is converted to `<li><p><picture></p></li>`. 

So I changed the condition to count list items with pictures as this really seems to be the root of the problem.